app/graphql/mutations/authentication/sign_up.rb

```diff
@@ -11,51 +11,15 @@ class SignUp < BaseMutation
       field :token, String, null: true
 
       def resolve(email:, password:, password_confirmation:, type:, referral_token: nil)
```

Comment:
@ManuSG8 
I've moved the logic to a `DryService` to clean up the mutation and better error handling